src/main/java/com/github/cameltooling/lsp/internal/parser/CamelYamlDSLParser.java

```diff
@@ -220,4 +254,46 @@ private String extractUriFromYamlData(String line) {
 		}
 		return null;
 	}
+
+	private int extractLines(TextDocumentItem textDocumentItem, Position position, String whenToStop,
+							 StringBuilder lines, int pos, boolean cleanNewLine) {
+		return extractLines(textDocumentItem, position, new String[]{whenToStop}, lines, pos, cleanNewLine);
+	}
+
+	private int extractLines(TextDocumentItem textDocumentItem, Position position, String[] whenToStopArray,
+							 StringBuilder lines, int pos, boolean cleanNewLine) {
+		// If it is a multi-line, cleanNewLine decides if we convert it to one very long single line
```

Comment:
Sometimes we want the YAML structure in multi-line. Sometimes we want one big string with no new lines in between. That's what cleanNewLine does.